### ISSUE TITLE
CLI interface

### ISSUE THREAD

**Issue by h2non** [vinci-proxy/vincix#4]
Command line parsers:
Go flag package (native)
Github.com/spf13/cobra

The command line interface allows to starts a custom vinci server based on the configuration file settings, passed as -config flag for instance.

**Comment by h2non** [comment]
@zzarcon or @groyoh, do you want to start working on this? It's something that you can start building from scratch. Then we can do code reviews to ensure we're covering the desired goals in the proper way.


Let me know if you need help in something. You can use other repositories as scaffold.

**Comment by groyoh** [comment]
Yeah I think I can dig into this. If I got it right, vincix simply parse the config and starts the vinci server(s) based on the config?

**Comment by h2non** [comment]
Exactly, that's the bottom line feature. The top goals is to provide a full proxy features, even with dynamic routing, monitoring and further instrumentation. Most of the existant repos will be imported by vincix in order to provide all those features. 

Also, the vincix name is open to discussion, but the unique bad part is it could be a confusing project name between vincix and vinci.

**Comment by groyoh** [comment]
This sounds awesome! Having vinci and vincix is indeed confusing, at least for me. But I'm pretty sure we can find a fancy and still meaningful name related to vinci. @zzarcon? :wink:

**Comment by zzarcon** [comment]
I really like the idea of having a cli for make easy the life of other devs. I'll just name it vinci-cli, just for avoid confusions ;)

**Comment by h2non** [comment]
Renamed. I think we can merge this repository into vinci as subpackage since Go will compile subpackages properly and generates the binary when found a main package. Let's see in a future.
